internal/handlers.go

```diff
@@ -13,7 +12,7 @@ const (
 func POSTHandler(next http.Handler) http.Handler {
 	fn := func(w http.ResponseWriter, req *http.Request) {
 		if req.Method != http.MethodPost {
-			http.Error(w, "required method POST", http.StatusMethodNotAllowed)
+			NewError(ErrorMethodNotAllowed, "required method POST").Print(w)
```

Comment:
Becomes `s.encodeError(w, NewFleetlockError(ErrorMethodNotAllowed, "required method POST"))`

Reply:
`s` (the `Server`) is not available here.

Reply:
Then have `encodeError` return be a standalone function, not a method. You've only made it a method to have that one fallback log at the end if encode fails in some way (which is a bug)

Reply:
Right. I do not really know when the json encoding can throw an error...